reflex/components/datadisplay/dataeditor.py

```diff
@@ -0,0 +1,353 @@
+    smoothScrollX: Var[bool]
+
+    # enable vertical smooth scrolling
+    smoothScrollY: Var[bool]
+
+    # Controls the drawing of the left hand vertical border of a column. If set to a boolean value it controls all borders.
+    # TODO: need to support a mapping (dict[int, bool] for handling separate columns)
+    verticalBorder: Var[bool]
+
+    columnSelect: Var[str]
+
+    #
+    preventDiagonalScrolling: Var[bool]
+
+    overscrollX: Var[int]
+
+    overscrollY: Var[int]
+
+    scrollOffsetX: Var[int]
+    scrollOffsetY: Var[int]
+
+    # global theme
+    theme: Var[dict]
+
+    is_default = True
+
+    def _get_imports(self):
+        return imports.merge_imports(
+            super()._get_imports(),
+            {
+                "": {ImportVar(tag=f"{self.library}/dist/index.css")},
+                self.library: {ImportVar(tag="GridCellKind")},
+                "/utils/helpers/dataeditor.js": {
+                    ImportVar(tag=f"getDEColumn", is_default=False, install=False),
+                    ImportVar(tag=f"getDERow", is_default=False, install=False),
+                    ImportVar(tag=f"locateCell", is_default=False, install=False),
+                    ImportVar(tag=f"formatCell", is_default=False, install=False),
+                    ImportVar(tag=f"onEditCell", is_default=False, install=False),
+                },
+            },
+        )
+
+    def get_event_triggers(self) -> Dict[str, Callable]:
+        """The event triggers of the component.
+
+        Returns:
+            The dict describing the event triggers.
+        """
+
+        def edit_sig(pos, data: dict[str, Any]):
+            return [pos, data]
+
+        return {
+            "onCellEdited": edit_sig,
+            "onGroupHeaderClicked": edit_sig,
+            "onGroupHeaderContextMenu": lambda grp_idx, data: [grp_idx, data],
+            "onCellActivated": lambda pos: [pos],
+            "onCellClicked": lambda pos: [pos],
+            "onCellContextMenu": lambda pos: [pos],
+            "onGroupHeaderRenamed": lambda idx, val: [idx, val],
+            "onHeaderClicked": lambda pos: [pos],
+            "onHeaderContextMenu": lambda pos: [pos],
+            "onHeaderMenuClick": lambda col, pos: [col, pos],
+            "onItemHovered": lambda pos: [pos],
+            "onDelete": lambda selection: [selection],
+            "onFinishedEditing": lambda new_value, movement: [new_value, movement],
+            "onRowAppended": lambda: [],
+            "onSelectionCleared": lambda: [],
+        }
+
+    def _get_hooks(self) -> str | None:
+        editor_id = get_unique_variable_name()
+        data_callback = f"getData_{editor_id}"
+        self.getCellContent = Var.create(data_callback, is_local=False)  # type: ignore
+
+        code = [f"function {data_callback}([col, row])" "{"]
+
+        code.extend(
+            [
+                f"  if (row < {self.data.full_name}.length && col < {self.columns.full_name}.length)"
```

Comment:
from this, it looks like `self.data` and `self.columns` can only be state vars, i.e passing values out of state will fail. Is this the intended behavior?